linter: react-hooks/exhaustive-deps - false positive setState

### What version of Oxlint are you using?

1.6.0

### What command did you run?

_No response_

### What does your `.oxlintrc.json` config file look like?

```jsonc
{
  "plugins": ["react-hooks"]
}
```


### What happened?

input with a false positive and a true positive:

```js
import { useEffect, useState } from "react";

export function Test() {
  const [state, setState] = useState();

  useEffect(() => {
    console.log("state", state); // false positive
  });

  useEffect(() => {
    setState(1); // true positive
  });
}
```

oxlint output - false positive then true positive:
```
  ⚠ eslint-plugin-react-hooks(exhaustive-deps): React Hook useEffect contains a call to setState. Without a list of dependencies, this can lead to an infinite chain of updates.
   ╭─[test.js:6:3]
 5 │
 6 │   useEffect(() => {
   ·   ─────────
 7 │     console.log("state", state);
   ╰────
  help: Consider adding an empty list of dependencies to make it clear which values are intended to be stable.

  ⚠ eslint-plugin-react-hooks(exhaustive-deps): React Hook useEffect contains a call to setState. Without a list of dependencies, this can lead to an infinite chain of updates.
    ╭─[test.js:10:3]
  9 │
 10 │   useEffect(() => {
    ·   ─────────
 11 │     setState(1);
    ╰────
  help: Consider adding an empty list of dependencies to make it clear which values are intended to be stable.

Found 2 warnings and 0 errors.
Finished in 5ms on 2 files using 16 threads.
```

equivalent eslint (v9.30.1) with `eslint.config.js`:

```js
import reactHooks from 'eslint-plugin-react-hooks';

export default [
  reactHooks.configs['recommended-latest'],
];
```

output - only true positive:
```
D:\scratch\oxlint-repro\test.js
  10:3  warning  React Hook useEffect contains a call to 'setState'. Without a list of dependencies, this can lead to an infinite chain of updates. To fix this, pass [] as a second argument to the useEffect Hook  react-hooks/exhaustive-deps

✖ 1 problem (0 errors, 1 warning)
```

--- a/crates/oxc_linter/src/rules/react/exhaustive_deps.rs
+++ b/crates/oxc_linter/src/rules/react/exhaustive_deps.rs
@@ -1408,15 +1408,32 @@ impl<'a> Visit<'a> for ExhaustiveDepsVisitor<'a, '_> {
         if let Some(decl) = get_declaration_of_variable(ident, self.semantic) {
             let is_set_state_call = match decl.kind() {
                 AstKind::VariableDeclarator(var_decl) => {
-                    if let Some(Expression::CallExpression(call_expr)) = &var_decl.init {
-                        if let Some(name) = func_call_without_react_namespace(call_expr) {
-                            name == "useState" || name == "useReducer"
-                        } else {
-                            false
-                        }
-                    } else {
-                        false
+                    let Some(Expression::CallExpression(call_expr)) = &var_decl.init else {
+                        return;
+                    };
+
+                    let Some(name) = func_call_without_react_namespace(call_expr) else {
+                        return;
+                    };
+
+                    if name != "useState" && name != "useReducer" {
+                        return;
                     }
+
+                    let BindingPatternKind::ArrayPattern(array_pat) = &var_decl.id.kind else {
+                        return;
+                    };
+
+                    let Some(Some(second_arg)) = array_pat.elements.get(1) else {
+                        return;
+                    };
+
+                    let BindingPatternKind::BindingIdentifier(binding_ident) = &second_arg.kind
+                    else {
+                        return;
+                    };
+
+                    binding_ident.name == ident.name
                 }
                 _ => false,
             };
@@ -2554,6 +2571,7 @@ fn test() {
         r"function MyComponent(props) { useEffect(() => { console.log(props.foo!.bar) }, [props.foo!.bar]) }",
         r"function MyComponent(props) { useEffect(() => { console.log((props.foo).bar) }, [props.foo!.bar]) }",
         r"function MyComponent(props) { const external = {}; const y = useMemo(() => { const z = foo<typeof external>(); return z; }, []) }",
+        r#"function Test() { const [state, setState] = useState(); useEffect(() => { console.log("state", state); }); }"#,
     ];
 
     let fail = vec![

